python_modules/dagster/dagster/_core/definitions/declarative_automation/operators/dep_operators.py

```diff
@@ -95,10 +95,11 @@ def ignore(self, selection: "AssetSelection") -> "DepCondition":
         )
         return self.model_copy(update={"ignore_selection": ignore_selection})
 
-    def _get_dep_keys(
-        self, asset_key: AssetKey, asset_graph: BaseAssetGraph
-    ) -> AbstractSet[AssetKey]:
-        dep_keys = asset_graph.get(asset_key).parent_keys
+    def _get_dep_keys(self, key: T_EntityKey, asset_graph: BaseAssetGraph) -> AbstractSet[AssetKey]:
```

Comment:
these changes make it valid for an AssetCheck to use any_deps_match

Reply:
It would be much more satisfying if this went to an underlying api that dealt with entity keys.

Reply:
@schrockn I updated an upstack PR to make this change (as we don't have the tools at this point in the stack to make that work): https://github.com/dagster-io/dagster/pull/24430/files#r1765409418